### DISCUSSION TITLE
containerd equivalent flag for allow_host_access

### DISCUSSION THREAD

**Discussion by bturner04** [concourse/concourse#6608] [General]
Hello,
We are current using the flag CONCOURSE_GARDEN_ALLOW_HOST_ACCESS.   Does containerd allow host access by default? Is there an equivalent flag that I will need to use?  

I do not see a flag currently in the worker spec file for containerd host access.

Ops-File:
path: /instance_groups/name=bosh/jobs/name=worker/properties/garden?/allow_host_access?
  value: true

Thanks!

**Comment by taylorsilva** [comment]
@muntac should this be an issue? Looking at the 7.0.0 release notes:

> If you rely on any Garden config that is not yet supported on our containerd runtime, please open an issue

**Reply by muntac** [reply]
Thanks. Yes, we can look into this. I'll open an issue.

**Reply by muntac** [reply]
@bturner04 I'm not very familiar with this Guardian feature. Curious what your use case for it looks like.

**Reply by bturner04** [reply]
@muntac 
We use a deployment that installs Concourse/Bosh Director/Credhub on to a single server.   
The concourse instance runs pipelines that communicates with the bosh director and credhub on the local host.   
Without the Guardian flag set to "true" the communication from concourse to bosh director is blocked.
Please let me know if you need more details.

The following links go into detail on why we had to start using this Guardian feature:
https://docs.pivotal.io/p-concourse/v6/advanced/garden-host-access/
https://github.com/concourse/concourse/issues/6294
 https://docs.pivotal.io/p-concourse/v6/rn/#breaking-changes

Thanks!

**Reply by muntac** [reply]
@bturner04 Thanks! I've opened an [issue](https://github.com/concourse/concourse/issues/6720) and added it to the current containerd related [epic](https://project.concourse-ci.org/projects/MDc6UHJvamVjdDExODQzMjcw).

**Reply by muntac** [reply]
@bturner04 after some investigation we found that we are likely allowing host access by default in containerd i.e. we don't set up the `REJECT` rule in iptables that Guardian does to prevent access.

We have some uncertainty about why exactly this works. Would you be able to do a test deployment in your setup with containerd and see if things work as intended?

**Reply by bturner04** [reply]
@muntac I tested this today with contrainerd as the default runtime in the manifest.  Concourse is working correctly and is able to communicate with the localhost.   Thank you for confirming that this access is allowed by default.
